RRS/accounts/tests.py

```diff
@@ -1,3 +1,177 @@
+
+    allauth_signals.user_signed_up.connect(
+        receiver=receivers.user_signed_up, sender='test')
+    allauth_signals.user_signed_up.send(
+        sender='test', request=None, user=user, social_login=social_login)
+
+    test_case.assertEqual(user.first_name, FIRST_NAME)
+    test_case.assertEqual(user.last_name, LAST_NAME)
+
+
+class SignUpTestCase(LinkSanityTestCase):
+
+    def setUp(self):
+        self.url_name = 'accounts:signup'
+        self.factory = RequestFactory()
+        self.user = User.objects.create_user(
+            'customer_username', 'customer@example.com', 'customer_password')
+        self.user.save()
+
+    def test__required_links__are_alive(self):
+        self.assert_valid_link(
+            expected_url='/signup/',
+            url_name=self.url_name,
+        )
+
+    def test__logged_in_user_redirect(self):
+        '''A logged in user should be redirected'''
+        assert_view_response_to_user(
+            self, self.url_name, self.user, status.HTTP_302_FOUND)
+
+    def test__anonymous_user_ok(self):
+        '''User that is not logged in should be able to view this page'''
+        assert_view_response_to_user(
+            self, self.url_name, AnonymousUser(), 200)
+
+    def test__form_complete_data(self):
+        complete_form_data = {'email': 'email@example.com',
+                              'first_name': 'firstname',
+                              'last_name': 'lastname',
+                              'password1': 'password',
+                              'password2': 'password'}
+
+        form = forms.SignUpForm(data=complete_form_data)
+        self.assertTrue(form.is_valid())
+
+    def test__form_missing_required_data(self):
+        required_form_fields = [
+            'email', 'first_name', 'last_name', 'password1', 'password2']
+
+        reverse_url = reverse(self.url_name)
+        response = self.client.post(reverse_url)
+
+        for field in required_form_fields:
+            self.assertFormError(
+                response, 'form', field, 'This field is required.')
+
+    def test__form_wrong_data(self):
+        wrong_form_data = {'email': 'not an email',
+                           'first_name': '',
+                           'last_name': '',
+                           'password1': 'password1',
+                           'password2': 'password2'}
+
+        form = forms.SignUpForm(data=wrong_form_data)
+        self.assertFalse(form.is_valid())
+
+    def test__user_signed_up_signal_facebook(self):
+        test_social_signed_up_signal(
+            test_case=self, provider=receivers.PROVIDER_FACEBOOK, user=self.user)
+
+    def test__user_signed_up_signal_google(self):
+        test_social_signed_up_signal(
+            test_case=self, provider=receivers.PROVIDER_GOOGLE, user=self.user)
+
+    def test__user_signed_up_signal_twitter(self):
+        test_social_signed_up_signal(
+            test_case=self, provider=receivers.PROVIDER_TWITTER, user=self.user)
```

Comment:
Would it be better if we use a single test case for social sign in, involving a list of possible providers?